repl/src/main/scala/ammonite/repl/interp/Evaluator.scala

```diff
@@ -197,18 +230,62 @@ object Evaluator{
       // Exhaust the printer iterator now, before exiting the `Catching`
       // block, so any exceptions thrown get properly caught and handled
       evaluatorRunPrinter(printer(evalMain(cls).asInstanceOf[Iterator[String]]))
+      evaluationResult(wrapperName, newImports)
+    }
+
+    def compileCacheBlock(wrapperName: String, code: String, scriptImports: Seq[ImportData]): Res[(Class[_], Seq[ImportData])] = for {
```

Comment:
the rest of the code (i.e. basically the manually-entered code) I don't care so much, but if it doesn't cost us anything and lets us eliminate code duplication, no reason we can't cache that too =D

At some point the code cache will grow big enough to be a problem and we'll need to expire things, but we can deal with that when the time comes